content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,861 @@
+stegen$pork[stegen$pork == 9] <- NA
+```
+
+let us break them down:
+
+1. `stegen$pork` means "get the variable called `pork` in the dataset `stegen`"
+2. `[...]` subset the vector according to `...`; if `...` is a series of
+   `TRUE/FALSE` values, only entries corresponding to `TRUE` are retained
+3. `... == 9` for each value of `...`  test if it is equal to `9`, and return `TRUE` if so (`FALSE` otherwise) 
+4. `... <- NA` replace `...` with `NA` (missing value)
+
+in other words: "isolate the entries of `stegen$pork` which equal `9`, and
+replace them with `NA`; here is another toy example to illustrate the procedure:
+
+```{r stegen_subset}
+## make toy input vector
+toy_vector <- 1:5
+toy_vector
+
+## make toy logical vector for subsetting
+toy_logical <- c(FALSE, TRUE, TRUE, FALSE, TRUE)
+toy_logical
+toy_vector[toy_logical] # subset values
+toy_vector[toy_logical] <- 0 # replace subset values
+toy_vector # check outcome
+
+```
+
+</details>
+
+## Saving clean data
+
+Now that we've cleaned our data, we will probably want to re-use it in further
+analyses, but we probably don't want to go through the steps of cleaning it
+again. Here, we will save our cleaned data into a new folder under `data/`
+called `cleaned/`. We will use **R**'s function `dir.create()` to create this
+folder under our data folder:
+
+```{r create-clean-dir, eval = FALSE}
+clean_dir <- here("data", "cleaned")
+dir.create(clean_dir)
+```
+
+After this, we can save our cleaned data into a new file called 
+`stegen_clean.csv`:
+
+```{r save-clean-data, eval = FALSE}
+stegen_clean_file <- here("data", "cleaned", "stegen_clean.csv")
+write.csv(stegen, file = stegen_clean_file, rownames = FALSE, sep = ",")
+```
+
+
+# Data exploration
+
+
+## Summaries of age and sex distribution
+
+We can have a brief look at age and sex distributions using some basic
+summaries; for instance:
+
+```{r stegen_age_sex}
+summary(stegen$age) # age stats
+summary(stegen$sex) # gender distribution
+tapply(stegen$age, INDEX = stegen$sex, FUN = summary) # age stats by gender      
+```
+
+
+
+<details>
+<summary> <b>Going further</b> click here to learn about <code>tapply()</code>: </summary>
+
+`tapply()` is a very handy function to stratify any kind of analyses. You will
+find more details by reading the documentation of the function `?tapply()`, but
+briefly, the syntax to be used is `tapply(input_data, stratification,
+function_to_use, optional_arguments)`. In the command used above:
+
+```{r eval = FALSE}
+tapply(stegen$age, INDEX = stegen$sex, FUN = summary)
+```
+
+this literally means: select the age variable in the dataset `stegen`
+(`stegen$age`), stratify it by sex (`stegen$sex`), and summaries each strata
+(`summary()`). So for instance, to get the average age by sex (function `mean()`), one could use:
+
+```{r stegen_age_per_sex}
+tapply(stegen$age, INDEX = stegen$sex, FUN = mean, na.rm = TRUE)
+```
+Here we illustrate that further arguments to the function can be passed as extra
```

Comment:
update "further arguments" here based on what you use above in function description